packages/cmf/src/bootstrap.js

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { render } from 'react-dom';
+import createSagaMiddleware from 'redux-saga';
+import { hashHistory } from 'react-router';
+import { routerMiddleware, syncHistoryWithStore } from 'react-router-redux';
+import { batchedSubscribe } from 'redux-batched-subscribe';
+import { call, fork } from 'redux-saga/effects';
+import compose from 'redux';
+
+import App from './App';
+import actionCreator from './actionCreator';
+import actions from './actions';
+import component from './component';
+import expression from './expression';
+import storeAPI from './store';
+import sagaRouter from './sagaRouter';
+import sagas from './sagas';
+import { registerInternals } from './register';
+
+const bactchedSubscribe = batchedSubscribe(notify => {
+	requestAnimationFrame(notify);
+});
+
+function assertTypeOf(options, attr, type) {
+	if (
+		type === 'Array' &&
+		options[attr] &&
+		!Array.isArray(options[attr]) &&
+		// eslint-disable-next-line valid-typeof
+		(options[attr] && typeof options[attr] !== type)
+	) {
+		throw new Error(`${attr} must be a ${type} but got ${typeof options[attr]}`);
+	}
+}
+
+/**
+ * This is the function to use in your app index.js file.
+ * It take your configuration and provide a very good default one.
+ * By default it start react with the following addons:
+ * - react-router
+ * - redux
+ * - redux-saga
+ * @param {object} options the set of supported options
+ * @returns {object} app object with render function
+ */
+export default function bootstrap(unSafeOptions) {
```

Comment:
please split, this function does to many differents things at once and is not really testable in a reliable way